src/domains.jl

```diff
@@ -69,6 +69,38 @@ upper(D::RealInterval) = D.upper
   (D.lower_open ? lower(D) < x : lower(D) ≤ x) && (D.upper_open ? x < upper(D) : x ≤ upper(D))
 end
 
+function Base.rand(rng::Random.AbstractRNG, d::RealInterval)
+  r = rand(rng, T)
+  low = lower(d)
+  upp = upper(d)
+  return r * (upp - low) + low
+end
+
+function Base.rand(rng::Random.AbstractRNG, d::RealInterval{T}) where {T <: AbstractFloat}
+  r = rand(rng, T)
+  bnd = 1/eps(T)
+  low = if lower(d) == -Inf || !d.lower_open
+    lower(d)
+  else
+    lower(d) + eps(T)
+  end
+  low = max(low, -bnd)
+  upp = if upper(d) == -Inf || !d.lower_open
+    upper(d)
+  else
+    upper(d) - eps(T)
+  end
+  upp = min(upp, bnd)
+  return r * (upp - low) + low
+end
+
+function Base.rand(rng::Random.AbstractRNG, d::RealInterval{T}) where {T <: Integer}
```

Comment:
Can you have a RealInterval that has a `Integer` type? it would make sense for the bounds, but then the value will also be forced to be an `Integer` and so it would just be the same as `IntegerRange`. Am I missing something?

Reply:
Agréed, this is not smart but possible as any `Real` is accepted. I think I added because of the unit tests

Reply:
  @testset "RealIntervals{$T}" verbose = true for T in TYPES